### DISCUSSION TITLE
how to use tailwind purge without html

### DISCUSSION THREAD

**Discussion by trashhalo** [tailwindlabs/tailwindcss#6063] [Help]
I have an edge case where I am trying to generate css for a proprietary 3rd party app that I can't control the html content.

They provide a theming engine where you can give it a css file to override some of the styling. Purge doesn't seem to work in this scenario and I wanted to know if I am doing something wrong.

```css
@tailwind base;
@tailwind components;
@tailwind utilities;

.foo {
  @apply text-green-900;
}
```

```js
module.exports = {
  purge: ["./src/**/*.css"],
  darkMode: false, // or 'media' or 'class'
  theme: {
    extend: {},
  },
  variants: {
    extend: {},
  },
  plugins: [],
  corePlugins: {
    preflight: false,
  },
};
```

## Output
Nothing purged.

## Expected output
Minimum required css to support the .foo apply and the .foo rules.

**Comment by michael-land** [comment]
you can use tailwind-safelist-generator to generate safelist file based on regex and then
```
  purge: ["./safelist.txt"],
```
or write a simple script extract classname from your css file and store in a txt file
